File: src/handlers.py
from vex import *
from definitions import brain, controller, driveTrain, belt, intake_flywheel

class Handlers:
    def __init__(self):
        # No need to pass anything in — we imported directly.
        self.brain = brain
        self.controller = controller
        self.driveTrain = driveTrain
        self.belt = belt
        self.intake = intake_flywheel

    # -----------------------
    # Generic Test function
    # -----------------------
    def test(self):
        self.controller.screen.print("Testing")
        self.brain.screen.print("Testing")

    # -----------------------
    # INTAKE EVENTS
    # -----------------------
    def intake_forward(self):
        self.intake.set_velocity(100, PERCENT)
        self.intake.spin(FORWARD)

    def intake_reverse(self):
        self.intake.set_velocity(100, PERCENT)
        self.intake.spin(REVERSE)

    def intake_stop(self):
        self.intake.stop()

    # -----------------------
    # BELT EVENTS
    # -----------------------
    def belt_forward(self):
        self.belt.set_velocity(80, PERCENT)
        self.belt.spin(FORWARD)

    def belt_reverse(self):
        self.belt.set_velocity(80, PERCENT)
        self.belt.spin(REVERSE)

    def belt_stop(self):
        self.belt.stop()

File: src/handlers.cpp
#include "handlers.h"

// Constructor
Handlers::Handlers() {
  // Leave empty unless you are initializing stuff
}

// --------------------------
// Generic
// --------------------------

void Handlers::test() {
  Controller1.Screen.print("Testing");
  Brain.Screen.print("Testing");
}

// --------------------------
// Intake controls
// --------------------------

void Handlers::intake_forward() {
  intake_flywheel.setVelocity(100, vex::percentUnits::pct);
  intake_flywheel.spin(vex::directionType::fwd);
}

void Handlers::intake_reverse() {
  intake_flywheel.setVelocity(100, vex::percentUnits::pct);
  intake_flywheel.spin(vex::directionType::rev);
}

void Handlers::intake_stop() {
  intake_flywheel.stop();
}

// --------------------------
// Belt controls
// --------------------------

void Handlers::belt_forward() {
  belt.setVelocity(80, vex::percentUnits::pct);
  belt.spin(vex::directionType::fwd);
}

void Handlers::belt_reverse() {
  belt.setVelocity(80, vex::percentUnits::pct);
  belt.spin(vex::directionType::rev);
}

void Handlers::belt_stop() {
  belt.stop();
}

// --------------------------
// Generic motor handler
// --------------------------

void Handlers::spinMotor(vex::motor& m, double speedPct) {
  if (fabs(speedPct) < 1) {
    m.stop(vex::brakeType::coast);
    return;
  }

  if (speedPct > 0) {
    m.spin(vex::directionType::fwd, speedPct, vex::percentUnits::pct);
  } else {
    m.spin(vex::directionType::rev, -speedPct, vex::percentUnits::pct);
  }
}
